src/main/java/org/spongepowered/common/mixin/core/world/TeleporterMixin.java

```diff
@@ -82,6 +85,27 @@ public void placeInPortal(Entity entityIn, float rotationYaw) {
         }
     }
 
+    @Override
+    public void bridge$placeEntity(net.minecraft.world.World world, Entity entity, float yaw) {
+        boolean didPort;
+        if (entity instanceof EntityPlayerMP) {
+            this.placeInPortal(entity, yaw);
+            didPort = true;
+        } else {
+            if (((WorldServerBridge) this.world).bridge$getDimensionId() == 1) {
```

Comment:
Is it guaranteed the world is a `WorldServer` instance?

Reply:
Yes

Reply:
Make the id a constant (Even though we know the end dimension id is `1` for all of 1.12's lifetime.